components/classroom/groups/checkPermissions.mjs

```diff
@@ -0,0 +1,30 @@
+import Permissions from "./permissions.mjs"
+
+function hasPermission(role, action, scope, entity){ //function checks if role definition meets permissions
+    if (!Permissions[role]) return false; //if role does not exist, return false
+
+    let rolePermissions = Permissions[role];
+
+    let permsArray = [];
+
+    for(let i=0; i<rolePermissions.length; i++){ //building a two-dimensional array (i is number of permissions, j is action-scope-entity)
```

Comment:
To clarify, is the reason why you are using the number of permissions is so you can iterate through the permissions the right number of times?